TransferMethod/Sources/View/Widget/TextWidget.swift

```diff
@@ -151,66 +144,58 @@ class TextWidget: AbstractWidget {
-                                                currentTextIndex: inout String.Index,
+    private func applyFormatForPatternCharacter(currentText: String,
                                                 finalText: inout String,
-                                                isEscapedCharacter: inout Bool,
+                                                index: inout CurrentIndex,
                                                 pattern: String,
-                                                patternIndex: inout String.Index,
                                                 patternCharactersToBeWritten: inout String) {
-        switch currentPatternCharacter.first {
+        let currentPatternCharacter = getCharacterAt(index.patternIndex, pattern)
+        let currentTextCharacter = getCharacterAt(index.textIndex, currentText)
+
+        switch currentPatternCharacter {
         case PatternCharacter.lettersAndNumbersPatternCharacter.rawValue:
-            formatTextForLettersAndNumbers(currentText: currentText,
-                                           currentTextCharacter: currentTextCharacter,
-                                           currentTextIndex: &currentTextIndex,
+            formatTextForLettersAndNumbers(currentTextCharacter: currentTextCharacter,
                                            finalText: &finalText,
+                                           index: &index,
                                            pattern: pattern,
-                                           patternCharactersToBeWritten: &patternCharactersToBeWritten,
-                                           patternIndex: &patternIndex)
+                                           patternCharactersToBeWritten: &patternCharactersToBeWritten)
+            index.textIndex = currentText.index(after: index.textIndex)
 
         case PatternCharacter.lettersOnlyPatternCharacter.rawValue,
              PatternCharacter.numbersOnlyPatternCharacter.rawValue:
             let filteredCharacter =
-                getTextForPatternCharacter(currentPatternCharacter.first!, currentTextCharacter)
-            formatTextForLettersAndNumbers(currentText: currentText,
-                                           currentTextCharacter: filteredCharacter,
-                                           currentTextIndex: &currentTextIndex,
-                                           finalText: &finalText,
-                                           pattern: pattern,
-                                           patternCharactersToBeWritten: &patternCharactersToBeWritten,
-                                           patternIndex: &patternIndex)
+                getTextForPatternCharacter(currentPatternCharacter, String(currentTextCharacter))
+            if let filteredCharacter = filteredCharacter, !filteredCharacter.isEmpty {
+                formatTextForLettersAndNumbers(currentTextCharacter: filteredCharacter.first!,
```

Comment:
You can make it simpler:
```
if let filteredCharacter = getTextForPatternCharacter(currentPatternCharacter,
                                                      String(currentTextCharacter)),
   let currentTextCharacter = filteredCharacter.first {
formatTextForLettersAndNumbers(currentTextCharacter: currentTextCharacter,
...
```

Reply:
It is not a good simplification for 2 reasons
1. filteredCharacter is String here and so we do need check for empty
2. We do need to pass filtered character and not currentTextCharacter (which was to be filtered)

Reply:
done